src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -586,17 +586,12 @@ static int add_partition_esp(DissectedPartition *p, bool has_xbootldr) {
          * Otherwise, if /efi/ is unused and empty (or missing), we'll take that.
          * Otherwise, we do nothing. */
         if (!has_xbootldr && slash_boot_exists()) {
-                r = slash_boot_in_fstab();
+                r = path_is_busy("/boot");
```

Comment:
This is wrong. If the ESP already has an entry in fstab, `fstab_has_node` above should handle it. If not, we should only use `/boot/` when it's not specified in fstab, and fallback to `/efi/` otherwise. You're just overmounting `/boot/` if it's empty, regardless of already being a mountpoint or not.

Reply:
Note that although `path_is_busy` checks whether the path is already a mountpoint, no mount unit will have been started by the time the generator does its initial run, i.e. during systemd startup. So we need to check fstab explicitly.

Reply:
Here is the `fstab` on my system:
```
UUID=93fe18d5-1331-4d58-b6a5-fceb1d130300	/         	ext4      	rw,relatime	0 1
UUID=1DAD-F136      	/boot     	vfat      	rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,utf8,errors=remount-ro	0 2 
```

Does this mean that `fstab_has_node` is not as expected?

Reply:
-> #28690